src/Common/ZooKeeper/ZooKeeperCommon.cpp

```diff
@@ -426,33 +435,66 @@ void ZooKeeperErrorResponse::writeImpl(WriteBuffer & out) const
     Coordination::write(error, out);
 }
 
+void ZooKeeperMultiRequest::checkOperationType(OperationType type)
+{
+    if (operation_type.has_value() && *operation_type != type)
+        throw Exception("Illegal mixing of read and write operations in multi request", Error::ZBADARGUMENTS);
```

Comment:
It means that our code tried to send an ill-formed request. Consider throwing `LOGICAL_ERROR` (but it will be `DB::Exception`, not `Coordination::Exception` then) or adding `chassert` here